### PR TITLE
Revise GatherTree reference implementation

### PR DESCRIPTION
**Pull request by ggalieroc**
### Details:
- [x] *nGraph op shell reviewed*
- [x] *Add visitor api unit test*
- [x] *Add template plugin test with reference values*
- [x] *Remove CoordinateTransform index() call from reference implementation (benchmark results in 45851)*
- [x] *Add serialization SLTs*
- [ ] *Add op to the list of trusted operations*

### Tickets:
 - *45851*, *45852*, *45853*, *53958*, *37513 (parent)*

### CODE REVIEW

File: docs/template_plugin/tests/functional/op_reference/gather_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+namespace {
+struct GatherTreeParams {
+    template <class IN_ET>
+    GatherTreeParams(const ngraph::Shape inShape, std::vector<IN_ET> stepIds, const std::vector<IN_ET> parentIds,
+        const std::vector<IN_ET> maxSeqLen, const std::vector<IN_ET> endToken, std::vector<IN_ET> output) :
+        stepIdsTensor(inShape, element::from<IN_ET>(), stepIds), parentIdsTensor(inShape, element::from<IN_ET>(), parentIds),
+        maxSeqLenTensor(ngraph::Shape{inShape[1]}, element::from<IN_ET>(), maxSeqLen), endTokenTensor(ngraph::Shape{}, element::from<IN_ET>(), endToken),
+        expectedTensor(inShape, element::from<IN_ET>(), output) {}
+Tensor stepIdsTensor;
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/gather_tree.cpp:28]: indent

Comment (ggalieroc) [reply docs/template_plugin/tests/functional/op_reference/gather_tree.cpp]: Done

**Review by bszmelcz — APPROVED** [review]

**Review by blesniewski — APPROVED** [review]
lgtm

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@ggalieroc Please rebase you branch to the latest master in order to avoid possible CI failures.

**Pull request merged**